### ISSUE TITLE
Chef Server Upgrade Migration 1.28 Fails During mover_server_admins_global_group_callback

### ISSUE THREAD

**Issue by phene** [chef/chef-server#1124]
```
[private-chef-upgrade] - Finished Migration 1.27 in 1.73 seconds
[private-chef-upgrade] - Current Migration Version: 1.24
[private-chef-upgrade] - Starting Migration 1.28
[private-chef-upgrade] - 	Stopping Services ["nginx", "opscode-erchef"]
ok: down: nginx: 314s, normally up
ok: down: opscode-erchef: 327s, normally up
[private-chef-upgrade] - Sleeping for 10 seconds while services stop...
[private-chef-upgrade] - 	Starting Services ["oc_bifrost", "postgresql"]
run: postgresql: (pid 14924) 281s; run: log: (pid 13286) 376s
[private-chef-upgrade] - Sleeping for 15 seconds while services comes online...
[private-chef-upgrade] - 	Restarting Service opscode-chef-mover
ok: run: opscode-chef-mover: (pid 15603) 0s
[private-chef-upgrade] - Sleeping for 15 seconds while service opscode-chef-mover comes back online...
run: opscode-chef-mover: (pid 15603) 18s; run: log: (pid 13620) 406s
[private-chef-upgrade] - Creating global group server-admins with read and create access on users container
[private-chef-upgrade] - 	Running Command: /opt/opscode/embedded/bin/escript /opt/opscode/embedded/service/opscode-chef-mover/scripts/migrate mover_server_admins_global_group_callback normal mover_transient_queue_batch_migrator with options {}
Unknown error error
 Unknown reason function_clause

=ERROR REPORT==== 7-Mar-2017::20:01:49 ===
{error,function_clause}

An unknown error has occurred.

bundler: failed to load command: ./bin/partybus (./bin/partybus)
RuntimeError: Partybus failed to execute command '/opt/opscode/embedded/bin/escript /opt/opscode/embedded/service/opscode-chef-mover/scripts/migrate mover_server_admins_global_group_callback normal mover_transient_queue_batch_migrator' exitstatus: 1
  /opt/opscode/embedded/service/partybus/lib/partybus/dsl_runner.rb:14:in `instance_eval'
  /opt/opscode/embedded/service/partybus/lib/partybus/migration_api/v1.rb:97:in `run_command'
  (eval):11:in `block in run'
  /opt/opscode/embedded/service/partybus/lib/partybus/migration_api/v1.rb:14:in `instance_eval'
  /opt/opscode/embedded/service/partybus/lib/partybus/migration_api/v1.rb:14:in `initialize'
  /opt/opscode/embedded/service/partybus/lib/partybus/dsl_runner.rb:25:in `new'
  /opt/opscode/embedded/service/partybus/lib/partybus/dsl_runner.rb:25:in `define_upgrade'
  (eval):1:in `run'
  /opt/opscode/embedded/service/partybus/lib/partybus/dsl_runner.rb:14:in `instance_eval'
  /opt/opscode/embedded/service/partybus/lib/partybus/dsl_runner.rb:14:in `run'
  /opt/opscode/embedded/service/partybus/lib/partybus/migrations.rb:41:in `run_migration'
  /opt/opscode/embedded/service/partybus/bin/partybus:62:in `block in do_upgrade'
  /opt/opscode/embedded/service/partybus/bin/partybus:58:in `each'
  /opt/opscode/embedded/service/partybus/bin/partybus:58:in `do_upgrade'
  /opt/opscode/embedded/service/partybus/bin/partybus:79:in `block in <top (required)>'
  /opt/opscode/embedded/service/partybus/bin/partybus:70:in `open'
  /opt/opscode/embedded/service/partybus/bin/partybus:70:in `<top (required)>'
```

`/var/log/opscode/opscode-chef-mover/error.log`:

```
2017-03-07 19:58:06.400 [error] 110111951111141033240110111951051004158Organization <<"elided-group-name">> failed during group addition.
2017-03-07 19:58:06.400 [error] 110111951111141033240110111951051004158Processing aborted in migrate due to error: {conn_failed,{error,econnrefused}}
2017-03-07 19:58:06.400 [error] 110111951111141033240110111951051004158in working: Worker down with unexpected error while validating.  Continuing validations. {error,{conn_failed,{error,econnrefused}},migrate}
2017-03-07 19:58:06.401 [error] 110111951111141033240110111951051004158gen_fsm <0.1357.0> in state migrate terminated with reason: {error,{conn_failed,{error,econnrefused}},migrate}
2017-03-07 19:58:06.401 [error] 110111951111141033240110111951051004158CRASH REPORT Process <0.1357.0> with 0 neighbours exited with reason: {error,{conn_failed,{error,econnrefused}},migrate} in gen_fsm:terminate/7 line 611
2017-03-07 19:58:06.401 [error] 110111951111141033240110111951051004158Supervisor mover_transient_worker_sup had child undefined started with {mover_transient_worker,start_link,undefined} at <0.1357.0> exit with reason {error,{conn_failed,{error,econnrefused}},migrate} in context child_terminated
2017-03-07 20:01:49.111 [error] 110111951111141033240110111951051004158Processing aborted in migrate due to error: {case_clause,{error,{conn_failed,{error,econnrefused}}}}
2017-03-07 20:01:49.111 [error] 110111951111141033240110111951051004158{migration_action,"Error in mover_server_admins_global_group_callback:migration_action"}
2017-03-07 20:01:49.112 [error] 110111951111141033240110111951051004158migration_action args
[<<"00000000000000000000000000000000">>,account_dets_not_opened]
2017-03-07 20:01:49.113 [error] 110111951111141033240110111951051004158error
{case_clause,{error,{conn_failed,{error,econnrefused}}}}
[{mv_oc_chef_authz,create_resource,2,[{file,"/var/cache/omnibus/src/opscode-chef-mover/_build/default/lib/mover/src/mv_oc_chef_authz.erl"},{line,378}]},{mover_server_admins_global_group_callback,create_server_admins_authz_group,1,[{file,"/var/cache/omnibus/src/opscode-chef-mover/_build/default/lib/mover/src/mover_server_admins_global_group_callback.erl"},{line,118}]},{mover_server_admins_global_group_callback,migration_action,2,[{file,"/var/cache/omnibus/src/opscode-chef-mover/_build/default/lib/mover/src/mover_server_admins_global_group_callback.erl"},{line,58}]},{mover_transient_worker,migrate,2,[{file,"/var/cache/omnibus/src/opscode-chef-mover/_build/default/lib/mover/src/mover_transient_worker.erl"},{line,42}]},{gen_fsm,handle_msg,7,[{file,"gen_fsm.erl"},{line,503}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,237}]}]
2017-03-07 20:01:49.113 [error] 110111951111141033240110111951051004158gen_fsm <0.1356.0> in state migrate terminated with reason: {error,{case_clause,{error,{conn_failed,{error,econnrefused}}}},migrate}
2017-03-07 20:01:49.113 [error] 110111951111141033240110111951051004158CRASH REPORT Process <0.1356.0> with 0 neighbours exited with reason: {error,{case_clause,{error,{conn_failed,{error,econnrefused}}}},migrate} in gen_fsm:terminate/7 line 611
2017-03-07 20:01:49.114 [error] 110111951111141033240110111951051004158in working: Worker down with unexpected error. Halting new workers. {error,{case_clause,{error,{conn_failed,{error,econnrefused}}}},migrate}
2017-03-07 20:01:49.114 [error] 110111951111141033240110111951051004158Supervisor mover_transient_worker_sup had child undefined started with {mover_transient_worker,start_link,undefined} at <0.1356.0> exit with reason {error,{case_clause,{error,{conn_failed,{error,econnrefused}}}},migrate} in context child_terminated
```

**Comment by phene** [comment]
Appears that oc_bifrost was not starting up fast enough.

**Comment by phene** [comment]
Adding a force-restart for `oc_bifrost` here[https://github.com/chef/chef-server/blob/master/omnibus/files/private-chef-upgrades/001/028_server_admins_global_group.rb#L8] fixed it for me.

**Comment by REWBrianH** [comment]
Experiencing this on upgrade from 12.4.0 to 13.1.13 - the force-restart bifrost change mentioned by @phene did not resolve the issue for me.

**Comment by REWBrianH** [comment]
My logs contain:
`_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}},{chef_secrets_app,start,[normal,[]]}} in application_master:init/4 line 134
2020-02-26 17:29:56.918 [error] 110111951111141033240110111951051004158Could not find CHEF_SECRETS_FD in environment
2020-02-26 17:29:56.919 [error] 110111951111141033240110111951051004158CRASH REPORT Process <0.1182.0> with 0 neighbours crashed with reason: no match of right hand value {error,env_var_not_found} in chef_secrets_keyring:init/1 line 43
2020-02-26 17:29:56.919 [error] 110111951111141033240110111951051004158Supervisor chef_secrets_sup had child chef_secrets_keyring started with chef_secrets_keyring:start_link([{provider,chef_secrets_fd}]) at undefined exit with reason no match of right hand value {error,env_var_not_found} in chef_secrets_keyring:init/1 line 43 in context start_error
2020-02-26 17:29:56.920 [error] 110111951111141033240110111951051004158CRASH REPORT Process <0.1179.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,chef_secrets_keyring,{{badmatch,{error,env_var_not_found}},[{chef_secrets_keyring,init,1,[{file,"/var/cache/omnibus/chef-server/src/opscode-chef-mover/_build/default/lib/chef_secrets/src/chef_secrets_keyring.erl"},{line,43}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,365}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,333}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}},{chef_secrets_app,start,[normal,[]]}} in application_master:init/4 line 134`

**Comment by ianau74** [comment]
What is opscode-chef-mover used for?  I'm seeing the same spammy error messages as @REWBrianH  in /var/log/opscode/opscode-chef-mover/error.log after upgrading to chef server core 13.1.13

2020-03-03 22:12:39.059 [error] 110111951111141033240110111951051004158Could not find CHEF_SECRETS_FD in environment
2020-03-03 22:12:39.059 [error] 110111951111141033240110111951051004158CRASH REPORT Process <0.1181.0> with 0 neighbours crashed with reason: no match of right hand value {error,env_var_not_found} in chef_secrets_keyring:init/1 line 43

I noticed that opscode-chef-mover was consuming a lot of CPU, trying to respawn this child process.

I've disabled opscode-chef-mover by adding this line to /etc/opscode/chef-server.rb
opscode_chef_mover['enable'] = false

and then running chef-server-ctl reconfigure clears up the errors.
